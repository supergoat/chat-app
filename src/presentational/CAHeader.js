// @flow
import React, { Component } from 'react';
import {
  Platform,
  Text,
  View
} from 'react-native';
import CAColors from "../common/CAColors";
import CAStyleSheet from "../common/CAStyleSheet";

/* Constants
============================================================================= */
const STATUS_BAR_HEIGHT = 20;
const IOS_HEADER_HEIGHT = 90 + STATUS_BAR_HEIGHT;
const ANDROID_HEADER_HEIGHT = 50 + STATUS_BAR_HEIGHT;

/* =============================================================================
<CAHeader />
--------------------------------------------------------------------------------
View Header

============================================================================= */
class CAHeader extends Component<{}, {}> {

  static __cards__: Function

  render() {
    return (
      <View style={styles.header}>
        <Text style={styles.title}>ChatApp</Text>
      </View>
    )
  }
}

/* StyleSheet
============================================================================= */
const styles = CAStyleSheet.create({
  header: {
    alignItems: 'center',
    justifyContent: 'center',
    alignItems: 'flex-start',
    padding: 15,
    ios: {
      paddingTop: IOS_HEADER_HEIGHT / 2,
      height: IOS_HEADER_HEIGHT,
      backgroundColor: CAColors.alabaster,
      borderBottomWidth: 1,
      borderBottomColor: CAColors.gallery,
    },
    android: {
      paddingTop: ANDROID_HEADER_HEIGHT / 2,
      height: ANDROID_HEADER_HEIGHT,
      backgroundColor: CAColors.riverBed,
    }
  },
  title: {
    color: CAColors.oxfordBlue,
    fontSize: 28,
    fontWeight: 'bold',
    android: {
      color: CAColors.white
    }
  }
});

/* Playground Cards
============================================================================= */
CAHeader.__cards__ = define => {
  define("Simple", () => <CAHeader />);
};


/* Export
============================================================================= */
export default CAHeader;
