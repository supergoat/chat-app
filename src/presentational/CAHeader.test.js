// @flow
import React from 'react';
import { View, Text } from 'react-native';
import CAHeader from './CAHeader';
import renderer from 'react-test-renderer';

describe('CAHeader', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<CAHeader />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when leftItem is provided', () => {
    const tree = renderer.create(
      <CAHeader
        leftItem={
          <View>
            <Text>left item</Text>
          </View>
        }
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when rightItem is provided', () => {
    const tree = renderer.create(
      <CAHeader
        rightItem={
          <View>
            <Text>right item</Text>
          </View>
        }
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when leftItem and rightItem are provided', () => {
    const tree = renderer.create(
      <CAHeader
        leftItem={
          <View>
            <Text>left item</Text>
          </View>
        }
        rightItem={
          <View>
            <Text>right item</Text>
          </View>
        }
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
