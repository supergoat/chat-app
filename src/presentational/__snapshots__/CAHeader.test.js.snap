// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CAHeader renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F9F9F9",
      "borderBottomColor": "#EEEEEE",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "height": 110,
      "justifyContent": "center",
      "padding": 15,
      "paddingTop": 55,
      "zIndex": 1,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#333C4E",
        "flex": 1,
        "fontSize": 28,
        "fontWeight": "bold",
      }
    }
  >
    ChatApp
  </Text>
</View>
`;

exports[`CAHeader renders correctly when leftItem and rightItem are provided 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F9F9F9",
      "borderBottomColor": "#EEEEEE",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "height": 110,
      "justifyContent": "center",
      "padding": 15,
      "paddingTop": 55,
      "zIndex": 1,
    }
  }
>
  <View
    style={
      Object {
        "height": "100%",
        "justifyContent": "center",
      }
    }
  >
    <View>
      <Text>
        left item
      </Text>
    </View>
  </View>
  <Text
    style={
      Object {
        "color": "#333C4E",
        "flex": 1,
        "fontSize": 28,
        "fontWeight": "bold",
      }
    }
  >
    ChatApp
  </Text>
  <View
    style={
      Object {
        "height": "100%",
        "justifyContent": "center",
      }
    }
  >
    <View>
      <Text>
        right item
      </Text>
    </View>
  </View>
</View>
`;

exports[`CAHeader renders correctly when leftItem is provided 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F9F9F9",
      "borderBottomColor": "#EEEEEE",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "height": 110,
      "justifyContent": "center",
      "padding": 15,
      "paddingTop": 55,
      "zIndex": 1,
    }
  }
>
  <View
    style={
      Object {
        "height": "100%",
        "justifyContent": "center",
      }
    }
  >
    <View>
      <Text>
        left item
      </Text>
    </View>
  </View>
  <Text
    style={
      Object {
        "color": "#333C4E",
        "flex": 1,
        "fontSize": 28,
        "fontWeight": "bold",
      }
    }
  >
    ChatApp
  </Text>
</View>
`;

exports[`CAHeader renders correctly when rightItem is provided 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F9F9F9",
      "borderBottomColor": "#EEEEEE",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "height": 110,
      "justifyContent": "center",
      "padding": 15,
      "paddingTop": 55,
      "zIndex": 1,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#333C4E",
        "flex": 1,
        "fontSize": 28,
        "fontWeight": "bold",
      }
    }
  >
    ChatApp
  </Text>
  <View
    style={
      Object {
        "height": "100%",
        "justifyContent": "center",
      }
    }
  >
    <View>
      <Text>
        right item
      </Text>
    </View>
  </View>
</View>
`;
